src/main/java/gov/nist/oar/rmm/repositories/impl/VersionReleasesetsRepositoryImpl.java

```diff
@@ -0,0 +1,157 @@
+package gov.nist.oar.rmm.repositories.impl;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.util.MultiValueMap;
+
+import com.mongodb.client.AggregateIterable;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Filters;
+
+import gov.nist.oar.rmm.config.AppConfig;
+import gov.nist.oar.rmm.config.MongoConfig;
+import gov.nist.oar.rmm.exceptions.ResourceNotFoundException;
+import gov.nist.oar.rmm.repositories.VersionReleasesetsRepository;
+import gov.nist.oar.rmm.utilities.ProcessRequest;
+
+@Service
+public class VersionReleasesetsRepositoryImpl implements VersionReleasesetsRepository {
+
+    private Logger logger = LoggerFactory.getLogger(VersionReleasesetsRepositoryImpl.class);
+
+    @Autowired
+    MongoConfig mconfig;
+
+    @Autowired
+    AppConfig appconfig;
+
+    /**
+     * Search Versions dataset to get version of the record This can be queried
+     * using key,value pairs as parameters
+     * 
+     * @param param key,value parameters
+     * @param p     pagination
+     * @return
+     */
+    @Override
+    public Document findVersion(MultiValueMap<String, String> param, Pageable p) {
+
+        return this.queryResults(param, p, "Versions");
+    }
+
+    /**
+     * Search Releasesets dataset using key,value pair where key is any field in the
+     * Releasets metadata schema
+     * 
+     * @param param
+     * @param p
+     * @return
+     */
+    @Override
+    public Document findReleaseset(MultiValueMap<String, String> param, Pageable p) {
+        return this.queryResults(param, p, "ReleaseSets");
```

Comment:
This hard-coded collection name does not match the hard-coded name in line 77.  Use instead the parameterized value from `MongoConfig`, `mconfig.releasesets`.

Reply:
This was changed some time ago I am not sure what code version is here. Also When I had created versioncontroller, I had not changed or updated Spring and mongo config. I need to test this part of code with the updates if you are merging it. I had noticed the null results in other sections too that's why I changed the repository implementation as well. Will take a look.